integration-tests/common-test-code/src/main/java/com/vaadin/flow/quarkus/it/routecontext/ChildNoOwnerView.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2000-2021 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.quarkus.it.routecontext;
+
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+import com.vaadin.flow.component.AttachEvent;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.NativeButton;
+import com.vaadin.flow.router.Route;
+import com.vaadin.flow.router.RouterLink;
+
+@Route(value = "child-no-owner", layout = ParentNoOwnerView.class)
+public class ChildNoOwnerView extends Div {
+
+    @Inject
+    private Instance<BeanNoOwner> instance;
```

Comment:
For this 'private' modifier my IDE gives a warning:
Private access modifier usage is not recommended for Quarkus CDI members.
```suggestion
   Instance<BeanNoOwner> instance;
```

Reply:
BTW, what is the purpose of using `Instance` here? Does the test imply that the instance of `BeanNoOwner` would vary during runtime?

Reply:
done

The purpose is the same as here https://github.com/vaadin/cdi/blob/master/vaadin-cdi-itest/src/main/java/com/vaadin/cdi/itest/routecontext/ChildNoOwnerView.java.
It's a direct copy and I don't want to go into this since it's the same code.

Injected `Instance<BeanNoOwner>` is not the same as injected `BeanNoOwner`  and the matter of this difference is exactly used in the test.
You can learn the `Instance<BeanNoOwner>` need and usage in CDI spec.